File: ABC100_199/ABC172/abc172/d.py
def f(n):
    divisors = []
    for i in range(1, int(n**0.5)+1):
        if n % i == 0:
            divisors.append(i)
            if i != n // i:
                divisors.append(n//i)

    return len(divisors)

if __name__ == "__main__":
  N = int(input())+1
  ans = [f(k) * k for k in range(N)]
  print(sum(ans))

File: ABC100_199/ABC172/abc172/d.cpp
#include <iostream>
using namespace std;
using ll = long long;

int main() {
  ll N;
  cin >> N;

  ll ans = 0;
  for (int i = 1; i <= N; i++){
      ll M = N/i;
      ans += (M*(M+1)/2)*i;
  }

  cout << ans << endl;
  return 0;
}